src/dotnet/Program.cs

```diff
@@ -138,6 +141,16 @@ internal static int ProcessArgs(string[] args, ITelemetry telemetryClient)
 
         }
 
+        /// <summary>
+        /// Performs initialization that must be done before anything else in the process.
+        /// </summary>
+        private static void InitializeProcess()
+        {
+            // by default, .NET Core doesn't have all code pages needed for Console apps.
+            // see the .NET Core Notes in https://msdn.microsoft.com/en-us/library/system.diagnostics.process(v=vs.110).aspx
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```

Comment:
As an FYI System.Text.Encoding.CodePages library is currently only in the shared framework because Roslyn pulls it in and it might not be there in the future which would cause it to be pulled app-local with the CLI because of this (perhaps there are already other dependencies on it as well). That assembly is ~750MB and is intentionally supposed to be pay-for-play, I'm not entirely sure why it is actually needed here as it should be needed to support Unicode encodings.